overlord/hookstate/hookmgr.go

```diff
@@ -41,6 +41,10 @@ type HookManager struct {
 	state      *state.State
 	runner     *state.TaskRunner
 	repository *repository
+
+	// RunTool is used to run the requested tool. It's exported here for testing
+	// purposes.
+	RunTool func(ToolRequest) (string, string)
```

Comment:
That's an attribute, exposed, inside the type, which doesn't even refer to the type. Quite unexpected.

Can we try having this as a normal method on the hook manager?  We can trivially test this by having a function called inside it for testing purposes, if we have to.